### PR TITLE
Added changes to PVC related tc to support KFNBC

### PR DESCRIPTION
**Pull request by tarukumar**
KFNBC PVC change and fixing debug error issue
Signed-off-by: Tarun Kumar <takumar@redhat.com>

### CODE REVIEW

**Comment by tarukumar** [comment]
https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/1270/. One TC failed ue to product bug

File: tests/Resources/Page/OCPDashboard/ConfigMaps/ConfigMaps.robot [hunk 1]
```diff
@@ -20,17 +20,17 @@ Check If ConfigMap Exists
     [Return]   ${status}
 
 Get PVC Size
-    [Documentation]    Get configure PVC size from configmap
-    [Arguments]   ${namespace}   ${configmap_name}=jupyterhub-cfg
-    ${data}    OpenShiftCLI.Get  kind=ConfigMap  namespace=${namespace}
+    [Documentation]    Get configure PVC size from OdhDashboardConfig CR
+    [Arguments]   ${namespace}   ${configmap_name}=odh-dashboard-config
+    ${data}    OpenShiftCLI.Get  kind=OdhDashboardConfig  namespace=${namespace}
```
Comment (jgarciao) [inline tests/Resources/Page/OCPDashboard/ConfigMaps/ConfigMaps.robot:25]: Could you use OpenShiftLibrary here?

File: tests/Resources/Page/OCPDashboard/ConfigMaps/ConfigMaps.robot [hunk 2]
```diff
@@ -20,17 +20,17 @@ Check If ConfigMap Exists
...
 Change PVC Size From ConfigMap
-    [Documentation]    Configure PVC size for JH
+    [Documentation]    Configure PVC size for Notebook controller
     ...    Supported size are whole number(ex: 120Gi,10Gi etc)
     ...    Decimal,alphabet charcter and number below 1 is not supported
-    [Arguments]   ${size}    ${configmap_name}=jupyterhub-cfg
-    OpenShiftCLI.Patch   kind=ConfigMap  name=${configmap_name}  namespace=${NAMESPACE}
-    ...    src={"data":{"singleuser_pvc_size": "${size}"}}
+    [Arguments]   ${size}    ${configmap_name}=odh-dashboard-config
+    OpenShiftCLI.Patch   kind=OdhDashboardConfig  name=${configmap_name}  namespace=${NAMESPACE}
```
Comment (jgarciao) [inline tests/Resources/Page/OCPDashboard/ConfigMaps/ConfigMaps.robot:35]: Could you use OpenShiftLibrary here?

File: setup.py [hunk 3]
```diff
@@ -16,7 +16,7 @@
     install_requires=[
         "reportportal-client",
         "robotframework==4.*",
-        "robotframework-debuglibrary",
+        "robotframework-debuglibrary>=2.0.0",
```
Comment (jgarciao) [inline setup.py:19]: Is this related to the error that we see in the logs recently?

**Review by jgarciao — CHANGES_REQUESTED** [review]

File: setup.py (hunk 3 above)
Comment (tarukumar) [reply setup.py:19]: yes

File: tests/Resources/Page/OCPDashboard/ConfigMaps/ConfigMaps.robot (hunk 2 above)
Comment (tarukumar) [reply tests/Resources/Page/OCPDashboard/ConfigMaps/ConfigMaps.robot:35]: Jorge i would rather change all the things together. There are other keyword which i am not using i am not sure where those keyword has been used. Rather changing for this keyword i would do while we do refactoring .

File: tests/Resources/Page/OCPDashboard/ConfigMaps/ConfigMaps.robot (hunk 1 above)
Comment (tarukumar) [reply tests/Resources/Page/OCPDashboard/ConfigMaps/ConfigMaps.robot:25]: Jorge i would rather change all the things together. There are other keyword which i am not using i am not sure where those keyword has been used. Rather changing for this keyword i would do while we do refactoring .

File: setup.py (hunk 3 above)
Comment (bdattoma) [reply setup.py:19]: shall we remove the library?

File: tests/Resources/Page/OCPDashboard/ConfigMaps/ConfigMaps.robot [hunk 4]
```diff
@@ -20,17 +20,17 @@ Check If ConfigMap Exists
...
 Get PVC Size
-    [Documentation]    Get configure PVC size from configmap
-    [Arguments]   ${namespace}   ${configmap_name}=jupyterhub-cfg
-    ${data}    OpenShiftCLI.Get  kind=ConfigMap  namespace=${namespace}
+    [Documentation]    Get configure PVC size from OdhDashboardConfig CR
+    [Arguments]   ${namespace}   ${configmap_name}=odh-dashboard-config
+    ${data}    OpenShiftCLI.Get  kind=OdhDashboardConfig  namespace=${namespace}
     ...    field_selector=metadata.name==${configmap_name}
-    ${size}    Set Variable      ${data[0]['data']['singleuser_pvc_size']}
+    ${size}    Set Variable      ${data[0]['spec']['notebookController']['pvcSize']}
```
Comment (bdattoma) [inline tests/Resources/Page/OCPDashboard/ConfigMaps/ConfigMaps.robot:27]: you could use the "fields" option of OpenshiftLibrary.Oc Get

**Review by jgarciao — APPROVED** [review]
lgtm

File: tests/Resources/Page/OCPDashboard/ConfigMaps/ConfigMaps.robot (hunk 1 above)
Comment (bdattoma) [reply tests/Resources/Page/OCPDashboard/ConfigMaps/ConfigMaps.robot:25]: I think it'd be safe enough to replace now this call

File: tests/Resources/Page/OCPDashboard/ConfigMaps/ConfigMaps.robot (hunk 2 above)
Comment (bdattoma) [reply tests/Resources/Page/OCPDashboard/ConfigMaps/ConfigMaps.robot:35]: I think it'd be safe enough to replace now this call

File: tests/Tests/400__ods_dashboard/411__ods_dashboard_settings_pvc_size.robot [hunk 5]
```diff
@@ -102,11 +102,13 @@ Verify Multiple Unsupported Size
...
-    [Documentation]    Make PVC change to configmap
-    ...    and rollout deployment config
+    [Documentation]    Make PVC change to OdhDashboardConfig CR
+    ...    and restart the Notebook controller POD
     [Arguments]     ${size}
     Check If PVC Change Is Permanent    ${size}
-    Roll Out Jupyter Deployment Config
+    Run Keyword And Ignore Error    Delete Pods Using Label Selector    redhat-ods-applications    app=notebook-controller
+    Wait Until Keyword Succeeds    15 times    1 sec   Verify Operator Pod Status    redhat-ods-applications    app=notebook-controller
+    Sleep    20
```
Comment (bdattoma) [inline tests/Tests/400__ods_dashboard/411__ods_dashboard_settings_pvc_size.robot:111]: did you try without restarting the pod? For examplen with KFNBC when you change the User Management settings you don't have to restart pods anymore, just wait kind of 10 seconds approx. Maybe it's the same with PVC (unless you tried)

Comment (tarukumar) [reply tests/Tests/400__ods_dashboard/411__ods_dashboard_settings_pvc_size.robot:111]: I have tried we need to wait. at least that what i saw in my verification

**Review by lugi0 — APPROVED** [review]

**Review by aloganat — APPROVED** [review]
lgtm

File: tests/Tests/400__ods_dashboard/411__ods_dashboard_settings_pvc_size.robot (hunk 5 above)
Comment (bdattoma) [reply tests/Tests/400__ods_dashboard/411__ods_dashboard_settings_pvc_size.robot:111]: so if we need to wait, why deleting the pods?

**Pull request merged**